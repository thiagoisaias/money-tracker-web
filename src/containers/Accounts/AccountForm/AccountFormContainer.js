import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import AccountForm from "../../../components/Accounts/AccountForm/AccountForm";

import { createAccount } from "../../../store/actions/accounts/accounts";

export class AccountFormContainer extends Component {
  submitData = formData => {
    const { userId, onCreateAccount } = this.props;
    
    onCreateAccount(formData, userId);
  };

  render() {
    const { error, isLoading } = this.props;
    return (
      <AccountForm
        submitData={this.submitData}
        error={error}
        isLoading={isLoading}
      />
    );
  }
}

AccountFormContainer.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  error: PropTypes.string,
  userId: PropTypes.number.isRequired,
  onCreateAccount: PropTypes.func.isRequired
};

const mapStateToProps = state => {
  return {
    userId: state.auth.user.id,
    isLoading: state.accounts.isLoading,
    error: state.accounts.error
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onCreateAccount: (accountData, userId) => {
      dispatch(createAccount(accountData, userId));
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(
  AccountFormContainer
);
