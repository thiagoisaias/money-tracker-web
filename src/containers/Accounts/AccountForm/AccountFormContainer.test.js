import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { AccountFormContainer } from "./AccountFormContainer";

describe("AccountFormContainer", () => {
  const mockProps = {
    accountToEdit: {
      id: 1,
      name: "Lorem",
      initialBalance: "25345.00"
    },
    error: null,
    history: {},
    isLoading: false,
    match: {},
    userId: 1,
    onClearAccountToEdit: jest.fn(),
    onCreateAccount: jest.fn(),
    onUpdateAccount: jest.fn(),
    onSetAccountToEdit: jest.fn()
  };
  const component = shallow(<AccountFormContainer {...mockProps} />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
