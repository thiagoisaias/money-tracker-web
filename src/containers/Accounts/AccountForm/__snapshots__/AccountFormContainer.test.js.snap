// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountFormContainer renders properly 1`] = `
<AccountForm
  error={null}
  id={1}
  initialBalance="25345.00"
  isLoading={false}
  match={Object {}}
  name="Lorem"
  submitData={[Function]}
/>
`;
